presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/OptimizeMixedDistinctAggregations.java

```diff
@@ -81,15 +87,19 @@ public OptimizeMixedDistinctAggregations(FunctionRegistry functionRegistry)
     @Override
     public Optional<PlanNode> apply(PlanNode node, Lookup lookup, PlanNodeIdAllocator idAllocator, SymbolAllocator symbolAllocator, Session session)
     {
+        if (!isOptimizeDistinctAggregationEnabled(session)) {
+            return Optional.empty();
+        }
+
         if (!(node instanceof AggregationNode)) {
             return Optional.empty();
         }
 
         AggregationNode aggregationNode = (AggregationNode) node;
 
         // optimize if and only if
-        // some aggregation functions have a distinct mask symbol
-        // and if not all aggregation functions on same distinct mask symbol (this case handled by SingleMarkDistinctToGroupBy)
+        // there is exactly one distinct aggregation
```

Comment:
its per column, and we use a `Set`. So this checks for distinct aggregations which are on the same column.